api/python/quilt3/backends/s3.py

```diff
@@ -13,8 +26,46 @@ def list_packages(self):
                 prev_pkg = pkg
                 yield pkg
 
-    def list_package_versions(self, pkg_name: str):
+    def list_package_pointers(self, pkg_name: str):
         package_dir = self.pointers_dir(pkg_name)
         for path, _ in list_url(package_dir):
             pkg_hash = get_bytes(package_dir.join(path))
             yield path, pkg_hash.decode().strip()
+
+    def delete_package(self, pkg_name: str):
+        delete_url_recursively(self.pointers_dir(pkg_name))
+
+
+class S3PackageRegistryV2(PackageRegistryV2):
```

Comment:
This could use a class comment describing the layout of the files in an S3 V2 registry.